staging/src/k8s.io/csi-translation-lib/plugins/in_tree_volume.go

```diff
@@ -88,23 +88,29 @@ func replaceTopology(pv *v1.PersistentVolume, oldKey, newKey string) error {
 	return nil
 }
 
-// getTopologyZones returns all topology zones with the given key found in the PV.
-func getTopologyZones(pv *v1.PersistentVolume, key string) []string {
+// getTopologyValues returns all unique topology values with the given key found in the PV.
+func getTopologyValues(pv *v1.PersistentVolume, key string) []string {
 	if pv.Spec.NodeAffinity == nil ||
 		pv.Spec.NodeAffinity.Required == nil ||
 		len(pv.Spec.NodeAffinity.Required.NodeSelectorTerms) < 1 {
 		return nil
 	}
 
-	var values []string
+	values := make(map[string]bool)
 	for i := range pv.Spec.NodeAffinity.Required.NodeSelectorTerms {
 		for _, r := range pv.Spec.NodeAffinity.Required.NodeSelectorTerms[i].MatchExpressions {
 			if r.Key == key {
-				values = append(values, r.Values...)
+				for _, v := range r.Values {
+					values[v] = true
```

Comment:
Because we will get duplicate values. For example, a PV has two NodeSelectorTerms and each of them has a region kubernetes topology which points to the same region.